client/src/reducers/config.ts

```diff
@@ -1,4 +1,6 @@
-export const config = (state = [], action) => {
+import { Action } from "../types/actions";
+
+export const config = (state = [], action: Action) => {
```

Comment:
Здесь фактическая ошибка: ты инициализируешь редьюсер литералом массива `[]`, а по факту у етебя объект.
```suggestion
export const config = (state: Config = {}, action: Action) => {
```

Если все поля Config сделать опциональными, либо проинициаизировать их